riscv/processor.h

```diff
@@ -354,88 +314,7 @@ class processor_t : public abstract_device_t
     HR_GROUP    /* Halt requested due to halt group. */
   } halt_request;
 
-  // Return the index of a trigger that matched, or -1.
-  inline int trigger_match(trigger_operation_t operation, reg_t address, reg_t data)
-  {
-    if (state.debug_mode)
-      return -1;
-
-    bool chain_ok = true;
-
-    for (unsigned int i = 0; i < state.num_triggers; i++) {
-      if (!chain_ok) {
-        chain_ok |= !state.mcontrol[i].chain;
-        continue;
-      }
-
-      if ((operation == OPERATION_EXECUTE && !state.mcontrol[i].execute) ||
-          (operation == OPERATION_STORE && !state.mcontrol[i].store) ||
-          (operation == OPERATION_LOAD && !state.mcontrol[i].load) ||
-          (state.prv == PRV_M && !state.mcontrol[i].m) ||
-          (state.prv == PRV_S && !state.mcontrol[i].s) ||
-          (state.prv == PRV_U && !state.mcontrol[i].u)) {
-        continue;
-      }
-
-      reg_t value;
-      if (state.mcontrol[i].select) {
-        value = data;
-      } else {
-        value = address;
-      }
-
-      // We need this because in 32-bit mode sometimes the PC bits get sign
-      // extended.
-      if (xlen == 32) {
-        value &= 0xffffffff;
-      }
-
-      auto tdata2 = state.tdata2->read(i);
-      switch (state.mcontrol[i].match) {
-        case MATCH_EQUAL:
-          if (value != tdata2)
-            continue;
-          break;
-        case MATCH_NAPOT:
-          {
-            reg_t mask = ~((1 << (cto(tdata2)+1)) - 1);
-            if ((value & mask) != (tdata2 & mask))
-              continue;
-          }
-          break;
-        case MATCH_GE:
-          if (value < tdata2)
-            continue;
-          break;
-        case MATCH_LT:
-          if (value >= tdata2)
-            continue;
-          break;
-        case MATCH_MASK_LOW:
-          {
-            reg_t mask = tdata2 >> (xlen/2);
-            if ((value & mask) != (tdata2 & mask))
-              continue;
-          }
-          break;
-        case MATCH_MASK_HIGH:
-          {
-            reg_t mask = tdata2 >> (xlen/2);
-            if (((value >> (xlen/2)) & mask) != (tdata2 & mask))
-              continue;
-          }
-          break;
-      }
-
-      if (!state.mcontrol[i].chain) {
-        return i;
-      }
-      chain_ok = true;
-    }
-    return -1;
-  }
-
-  void trigger_updated();
+  void trigger_updated(const std::vector<triggers::trigger_t *> *triggers);
```

Comment:
I think the preferred C++ way here is to use const ref instead of passing a pointer to vector.

Reply:
I'll change it.